src/Stl.Interception/IMethodBasedFactory.cs

```diff
@@ -0,0 +1,3 @@
+namespace Stl.Interception;
+
+public interface IMethodBasedFactory;
```

Comment:
Should curiously recurring template pattern be used here?
I.e. `IMethodBasedFactory<T> where T : IMethodBasedFactory<T>, IRequiresAsyncProxy`.

Reply:
I would say no - I usually do this only when it's really necessary, i.e. when you end up looking for this `T` parameter in the inheritance chain.